src/lib.rs

```diff
@@ -145,10 +149,63 @@ mod tests {
     // #[serial]
     // fn test_display_address_with_path_taproot() {}
 
-    // TODO: Create PSBT with scratch using given Hardware Wallet
-    // #[test]
-    // #[serial]
-    // fn test_sign_tx() {}
+    #[test]
+    #[serial]
+    fn test_sign_tx() {
+        let devices = HWIClient::enumerate().unwrap();
+        let device = devices.first().unwrap();
+        let client = HWIClient::get_client(device, true, types::HWIChain::Test).unwrap();
+        let derivation_path = DerivationPath::from_str("m/44'/1'/0'/0/0").unwrap();
+
+        let address = client
+            .display_address_with_path(&derivation_path, types::HWIAddressType::Legacy)
+            .unwrap();
+
+        let pk = client.get_xpub(&derivation_path, true).unwrap();
+        let mut hd_keypaths: BTreeMap<secp256k1::PublicKey, KeySource> = Default::default();
+        // Here device fingerprint is same as master xpub fingerprint
+        hd_keypaths.insert(pk.public_key, (device.fingerprint, derivation_path));
+
+        let tx = Transaction {
+            version: 1,
+            lock_time: 0,
+            input: vec![TxIn::default()],
+            output: vec![TxOut {
+                value: 100,
+                script_pubkey: address.address.script_pubkey(),
+            }],
+        };
+
+        let txin = TxIn {
```

Comment:
I would call this "previous_txin", so it's easier to understand that it's not the current tx's txin